src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -99,22 +99,19 @@ internal class WorkloadInstallCommand : CommandBase
             ValidateWorkloadIdsInput();
         }
 
-        private void ValidateWorkloadIdsInput()
+        void ValidateWorkloadIdsInput()
         {
             var availableWorkloads = _workloadResolver.GetAvailableWorkloads();
             foreach (var workloadId in _workloadIds)
             {
-                if (!availableWorkloads.Select(workload => workload.Id.ToString()).Contains(workloadId))
-                {
-                    if (_workloadResolver.IsPlatformIncompatibleWorkload(new WorkloadId(workloadId)))
-                    {
-                        throw new GracefulException(string.Format(LocalizableStrings.WorkloadNotSupportedOnPlatform, workloadId), isUserError: false);
-                    }
-                    else
-                    {
-                        throw new GracefulException(string.Format(LocalizableStrings.WorkloadNotRecognized, workloadId), isUserError: false);
-                    }
-                }
+                if (availableWorkloads.Any(workload => workload.Id.ToString() == workloadId))
+                    continue;
```

Comment:
This allocates an extra closure and a non-static delegate, but perhaps fewer enumerators. I don't know whether efficiency is important here.

Reply:
Should be doing a lot less enumerations because Any will break ASAP.